pkg/signals/signals_test.go

```diff
@@ -0,0 +1,82 @@
+//go:build !windows
+// +build !windows
+
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signals
+
+import (
+	"os"
+	"syscall"
+	"testing"
+	"time"
+)
+
+var (
+	settleTime       = 100 * time.Millisecond
+	fatalWaitingTime = 15 * time.Second
+)
+
+func waitSig(t *testing.T, c <-chan os.Signal, sig os.Signal) {
+	t.Helper()
+	timeout := time.After(fatalWaitingTime)
+	ticker := time.Tick(settleTime)
+	for {
+		select {
+		case <-timeout:
+			t.Fatalf("Timeout after %v waiting for signal", fatalWaitingTime)
+		case <-ticker:
+			if len(c) != 0 {
+				s := <-c
+				if s == sig {
+					return
+				}
+			}
+		}
```

Comment:
```suggestion
		select {
		case <-timeout:
			t.Fatalf("Timeout after %v waiting for signal", fatalWaitingTime)
		case s:= <- c:
		        if s == sig {
		                return
		        }
		}
```
Just understand what the test wants to do, it could just use a select with two cases, no another ticker or timer is required. I think the timeout could be much shorter.

Reply:
Refined, thanks.